src/fasta-recs.c

```diff
@@ -9,8 +40,63 @@ int main(int argc, char const *argv[])
     }
 
     const char *fasta_fname = argv[1];
-    printf("I should be parsing %s now, but someone forgot to implement that.\n",
-           fasta_fname);
+
+    char *content = readFile(fasta_fname);
+
+    //Parsing into DNA chains
+    int pos = 0;
+    const int length = strlen(content);
+
+    // Find every chain
+    while(pos<length){
+        //First we have to find '>'
+        char c = content[pos];
+        while(pos < length && c!='>'){
+            pos++;
+            c = content[pos];
+        }
+
+        // Skip spaces
+        while(pos < length && c!=' '){
+            pos++;
+            c = content[pos];
+        }
+
+        // Find the name
+        int nameBegin = pos;
+        while(pos<length && c!=' ' && c!='\n'){
+            pos++;
+            c = content[pos];
+        }
+
+        char *name = calloc(pos-nameBegin, sizeof(char));
```

Comment:
`sizeof(char)` is always 1, by definition, so we usually just write 1. Not that there is anything wrong with `sizeof(char)`. If you use `sizeof(*name)` instead, you get the size of the elements in the result automatically. That is safer in general.

If the string you want to put in `name` is `n` long, you need to allocate `n+1` bytes. You need to termiante the string. I don't think you do that here.